### DISCUSSION TITLE
Filament Change notification (Multi-Color Print)

### DISCUSSION THREAD

**Discussion by tekierz** [greghesp/ha-bambulab#610] [Q&A]
Is there a way to determine if the toolhead is moving to the side because the filament is color is changing or because of the smooth Timelapse mode?

 I've mounted a limit switch and connected it to my camera (via a rasp pi). 

I've integrated HA status via 'sensor.a1_03919c452109073_current_stage'. 

However I'm not sure how to distinguish between a filament change and a timelapse photo.

**Comment by AdrianGarside** [comment]
Filament changes can be detected when the active_tray sensor changes content.

**Comment by tekierz** [comment]
Oh ok, so I finally figured it out by just dumping all the logs, so what you actually want to look for is for when the "active_tray" is unavailable ((this maybe be dependent on the model))

Alternatively, is there a way to get the gcode from HA / pybambu?

**Comment by AdrianGarside** [comment]
active_tray going unavailable shouldn't be happening during a print - unless it does mid color change?

**Comment by tekierz** [comment]
@AdrianGarside yeah, it temporarily become unavailable during filament changes, which is what I was looking for.  I just need an indication of mid-layer filament change in order to get my script to support multicolor timelapses
